### PR TITLE
x-pack/filebeat/input/entityanalytics/provider/okta/internal/okta: new package

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

This adds a new package for accessing the Okta user API.

## Why is it important?

It is necessary for adding the Okta entity analytics input.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

Obtain an Okta dev account and set up an auth token, then you can run the tests in x-pack/filebeat/input/entityanalytics/provider/okta/internal/okta with
```
OKTA_HOST=<oktadomain> OKTA_TOKEN=<token> go test -v -log_user_response
```

## Related issues

- For elastic/security-team#5100

### CODE REVIEW

**Comment by jamiehynds** [comment]
@taylor-swanson would you mind reviewing this one? Dan has added support for Okta users to the input, and Crest hoping to start on the package itself once we get this PR reviewed & merged.

**Comment by efd6** [comment]
@jamiehynds This is the first half, the actual input will build on this.

File: x-pack/filebeat/input/entityanalytics/provider/okta/internal/okta/okta.go [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+	Name *string `json:"name,omitempty"`
+}
+
+// Response is a set of omit options specifying a part of the response to omit.
+//
+// See https://developer.okta.com/docs/reference/api/users/#content-type-header-fields-2 for details.
+type Response uint8
+
+const (
+	// Omit the credentials sub-object from the response.
```
Comment (taylor-swanson) [inline x-pack/filebeat/input/entityanalytics/provider/okta/internal/okta/okta.go:112]: My editor flags these docstrings as not beginning with the variable name (`// OmitCredentials omits ...`. I'm not sure if this is something we enforce (the linter doesn't seem to care), but thought I'd mention it.

**Review by taylor-swanson — APPROVED** [review]
This looks really great! Aside from a minor comment and fixing the linter issues, LGTM.

File: x-pack/filebeat/input/entityanalytics/provider/okta/internal/okta/okta.go (hunk 1 above)
Comment (efd6) [reply x-pack/filebeat/input/entityanalytics/provider/okta/internal/okta/okta.go:112]: I'm OK with this break in style. The intention is clear.

**Pull request merged**